### DISCUSSION TITLE
Wrapper for searching for multiple substrings?

### DISCUSSION THREAD

**Discussion by rieje** [BurntSushi/ripgrep#2765] [Q&A]
I often find myself searching for media file names in the following way: 

`rg 'cat.*videos . ; rg videos.*cat .` where I'm interested in searching the substrings `videos` and `cat` in any order. I might have multiple substrings.

Can rg/regex help do that in any way? Otherwise I have to write a wrapper script to do that but my posix/bash/programming is not great. I guess it would be something like this:

`./rg-substrings cat videos` where it loops through each argument and appends a `.*` that gets added to the resulting regex string for rg. But the harder part is looping through all iterations of the orders of the substring.

**Comment by BurntSushi** [comment]
If I'm understanding you correctly, then the standard way to do this is with shell pipelines: `rg cat | rg videos`.

**Reply by rieje** [reply]
I see, it's not possible to show the colors of both `cat` and `videos` that get matched if done that way, i.e. only `videos`, right?

Also to be sure logical AND for patterns is not supported?

I currently have in a shell script:

    regex=""

    for f in "$@"; do
      regex+="$f.*"
    done

    rg "$regex" ~/log

but thinking about how to implement logical AND to produce result lines that contain all substrings without respecting the order the substrings were given as arguments.

**Reply by BurntSushi** [reply]
See: https://github.com/BurntSushi/ripgrep/issues/875

**Reply by BurntSushi** [reply]
For colors: `rg cat --color always | rg videos`.
